vault/plugin_catalog.go

```diff
@@ -63,10 +70,29 @@ type MultiplexedClient struct {
 	// name is the plugin name
 	name string
 
-	client *plugin.Client
+	// connections holds client connections by ID
+	connections map[string]*PluginClient
+}
+
+func (c *Core) setupPluginCatalog(ctx context.Context) error {
```

Comment:
Will we ever return an error?

Reply:
resolved in: https://github.com/hashicorp/vault/pull/13896/commits/8adfd959af3486c89e46e5d58593608759ed11d1